akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -360,6 +360,60 @@ private[akka] final case class Fold[In, Out](zero: Out, f: (Out, In) ⇒ Out, de
 /**
  * INTERNAL API
  */
+final class FoldAsync[In, Out](zero: ⇒ Out, f: (Out, In) ⇒ Future[Out]) extends GraphStage[FlowShape[In, Out]] {
+  import akka.dispatch.ExecutionContexts
+
+  val in = Inlet[In]("FoldAsync.in")
+  val out = Outlet[Out]("FoldAsync.out")
+
+  val shape = FlowShape.of(in, out)
+
+  def createLogic(inheritedAttributes: Attributes): GraphStageLogic =
+    new GraphStageLogic(shape) {
+      private var aggregated: Out = zero
+      private var aggregating: Future[Out] = Future.successful(aggregated)
+
+      private def handle(output: Try[Out]): Unit = output match {
+        case Failure(reason) ⇒ fail(out, reason)
+        case Success(o) ⇒
+          aggregated = o
+          if (!isClosed(in)) pull(in)
+      }
+
+      implicit private def ec = ExecutionContexts.sameThreadExecutionContext
```

Comment:
Actually I've tested a variant with the EC resolved from the implicit scope when built, but as the `FutureStage` seems to use that, I was not sure